qupath-gui-fx/src/main/java/qupath/lib/gui/viewer/QuPathViewer.java

```diff
@@ -2749,28 +2758,38 @@ protected void updateAffineTransform() {
 	 * @param theta
 	 */
 	public void setRotation(double theta) {
```

Comment:
This can fail when having two viewers, turning off synchronization, rotating one a little, and then turning synchronization back on. Subsequently when rotating around a full circle, at one point the other viewer will be 'locked' in position until being released after a threshold rotation is reached.

I think the following method works (not very tested!)
```java
        public void setRotation(double theta) {
		if (rotationProperty.get() == theta)
			return;
		while (theta < 0)
			theta += 2 * Math.PI;
		theta = theta % (2 * Math.PI);
		rotationProperty.set(theta);
	}
```